carp.data.core/src/commonMain/kotlin/dk/cachet/carp/data/application/DataStreamServiceHost.kt

```diff
@@ -0,0 +1,29 @@
+package dk.cachet.carp.data.application
+
+import dk.cachet.carp.common.application.UUID
+import dk.cachet.carp.data.domain.DataStreamRepository
+
+
+/**
+ * Store and retrieve [DataStreamPoint]s for study deployments.
+ */
+class DataStreamServiceHost( private val repository: DataStreamRepository ) : DataStreamService
```

Comment:
Not sure if this is to be regarded as an edge service or an internal service receiving data through different channels. If this is an edge service is should probably not have a repository, but something more flexible like a "sink" interface that can have any implementation required by the underlying infrastructure.